### ISSUE TITLE
404 on icons request after upgrading to M5

### ISSUE THREAD

**Issue by junalmeida** [openhab/openhab-webui#1092]
## The problem

Icon requests are failing after upgrading to latest 3.1.0M5

![image](https://user-images.githubusercontent.com/991675/120893847-ee826f00-c5da-11eb-86e4-356a99a4ce86.png)

## Expected behavior

Icons to appear when using openHAB 3 ui

## Steps to reproduce

1. Open locations tab
2. Open any location with items with icons

## Your environment

```yaml
runtimeInfo:
  version: 3.1.0.M5
  buildString: Milestone Build
locale: en-CA
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 11.0.11
  javaVendor: Azul Systems, Inc.
  javaVendorVersion: Zulu11.48+21-CA
  osName: Linux
  osVersion: 5.10.17-v8+
  osArchitecture: aarch64
  availableProcessors: 4
  freeMemory: 58172096
  totalMemory: 194838528
bindings:
  - deconz
  - nest
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1
    prefersColorScheme: dark
  isSecureContext: false
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 8
    language: en-US
    languages:
      - en-US
      - en
      - pt
      - pt-BR
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1080
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: dark
    filled: true
    pageTransitionAnimation: default
    bars: light
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
    like Gecko) Chrome/91.0.4472.77 Safari/537.36 Edg/91.0.864.37
timestamp: 2021-06-05T13:51:16.166Z

```

## Browser console

```txt
door:1 GET http://smarthome:8080/icon/door?format=svg&anyFormat=true&state=open 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
_e @ app.js:7
setItemState @ app.js:38
(anonymous) @ app.js:17
(anonymous) @ app.js:23
(anonymous) @ app.js:23
u._withCommit @ app.js:23
u.commit @ app.js:23
commit @ app.js:17
(anonymous) @ app.js:38
r.onmessage @ app.js:33
heating:1 GET http://smarthome:8080/icon/heating?format=svg&anyFormat=true&state=OFF 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:23
setTimeout (async)
cardOpening @ app.js:23
qe @ app.js:7
n @ app.js:7
qe @ app.js:7
e.$emit @ app.js:7
(anonymous) @ app.js:7
t.a @ app.js:7
dispatchEvent @ app.js:23
onOpened @ app.js:23
(anonymous) @ app.js:7
(anonymous) @ app.js:7
value @ app.js:7
X @ app.js:7
(anonymous) @ app.js:7
r @ app.js:1
c @ app.js:1
fire:1 GET http://smarthome:8080/icon/fire?format=svg&anyFormat=true&state=OFF 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
actualState @ app.js:33
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
_e @ app.js:7
setItemState @ app.js:38
(anonymous) @ app.js:17
(anonymous) @ app.js:23
(anonymous) @ app.js:23
u._withCommit @ app.js:23
u.commit @ app.js:23
commit @ app.js:17
(anonymous) @ app.js:38
r.onmessage @ app.js:33
SoundVolume:1 GET http://smarthome:8080/icon/SoundVolume?format=svg&anyFormat=true&state=21 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
actualState @ app.js:33
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
_e @ app.js:7
setItemState @ app.js:38
(anonymous) @ app.js:17
(anonymous) @ app.js:23
(anonymous) @ app.js:23
u._withCommit @ app.js:23
u.commit @ app.js:23
commit @ app.js:17
(anonymous) @ app.js:38
r.onmessage @ app.js:33
ColorLight:1 GET http://smarthome:8080/icon/ColorLight?format=svg&anyFormat=true 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
click @ app.js:23
qe @ app.js:7
n @ app.js:7
qe @ app.js:7
e.$emit @ app.js:7
(anonymous) @ app.js:7
t.a @ app.js:7
dispatchEvent @ app.js:23
onClick @ app.js:23
Temperature:1 GET http://smarthome:8080/icon/Temperature?format=svg&anyFormat=true 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
click @ app.js:23
qe @ app.js:7
n @ app.js:7
qe @ app.js:7
e.$emit @ app.js:7
(anonymous) @ app.js:7
t.a @ app.js:7
dispatchEvent @ app.js:23
onClick @ app.js:23
Humidity:1 GET http://smarthome:8080/icon/Humidity?format=svg&anyFormat=true 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
click @ app.js:23
qe @ app.js:7
n @ app.js:7
qe @ app.js:7
e.$emit @ app.js:7
(anonymous) @ app.js:7
t.a @ app.js:7
dispatchEvent @ app.js:23
onClick @ app.js:23
heating:1 GET http://smarthome:8080/icon/heating?format=svg&anyFormat=true&state=OFF 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
click @ app.js:23
qe @ app.js:7
n @ app.js:7
qe @ app.js:7
e.$emit @ app.js:7
(anonymous) @ app.js:7
t.a @ app.js:7
dispatchEvent @ app.js:23
onClick @ app.js:23
SoundVolume:1 GET http://smarthome:8080/icon/SoundVolume?format=svg&anyFormat=true&state=21 404 (Not Found)
Image (async)
ga @ app.js:7
va @ app.js:7
ma @ app.js:7
S @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
(anonymous) @ app.js:33
Promise.then (async)
updateIcon @ app.js:33
mounted @ app.js:33
qe @ app.js:7
tn @ app.js:7
insert @ app.js:7
O @ app.js:7
(anonymous) @ app.js:7
e._update @ app.js:7
a @ app.js:7
fn.get @ app.js:7
fn.run @ app.js:7
pn @ app.js:7
(anonymous) @ app.js:7
Ze @ app.js:7
Promise.then (async)
Ye @ app.js:7
at @ app.js:7
(anonymous) @ app.js:7
fn.update @ app.js:7
pe.notify @ app.js:7
set @ app.js:7
mn.set @ app.js:7
click @ app.js:23
qe @ app.js:7
n @ app.js:7
qe @ app.js:7
e.$emit @ app.js:7
(anonymous) @ app.js:7
t.a @ app.js:7
dispatchEvent @ app.js:23
onClick @ app.js:23
fire:1 GET http://smarthome:8080/icon/fire?format=svg&anyFormat=true&state=OFF 404 (Not Found)
```

## Browser network traffic

![image](https://user-images.githubusercontent.com/991675/120894138-8fbdf500-c5dc-11eb-8ad1-4e980f39fb07.png)

![image](https://user-images.githubusercontent.com/991675/120894164-a5cbb580-c5dc-11eb-9d57-c1ed28b76525.png)

**Comment by hubsif** [comment]
Hi @junalmeida,
I just had the same issue, Firefox's network console would also show 404 errors for me. I then did a hard refresh (Shift-Ctrl/Cmd-R) and then the icons loaded.

It seems to be a (Firefox-) caching issue. Though, it seems weird to me that it reported 404 replies. Unfortunately, after the hard refresh I can't reproduce it anymore. And one of the failing network request I copied as curl command works just fine on the console 🤔.
